### PR TITLE
ci: adapt golang 1.19.2

### PR DESCRIPTION
**Pull request by bergwolf**
* drop `go get`
* drop `godep`
* update cni config

Depends-on: github.com/kata-containers/kata-containers#5511
Fixex: #5223

### CODE REVIEW

**Comment by bergwolf** [comment]
//test-ubuntu

File: .ci/static-checks.sh [hunk 1]
```diff
@@ -915,41 +915,9 @@ static_check_vendor()
...
-		fi
-	fi
-
-	info "Checking vendoring metadata"
-
-	# Get the vendoring tool
-	go get github.com/golang/dep/cmd/dep
-
-	# Check, but don't touch!
-	dep check
```
Comment (littlejawa) [inline .ci/static-checks.sh:952]: This block removal seems unrelated to the commit. Am I missing something?

Comment (bergwolf) [reply .ci/static-checks.sh:952]: It is, sort of. The PR is to drop `go get` in order to be able to upgrade golang version (see the depends-on tag), which also invalidates the use of `dep`. Let me split it into another commit.

**Review by littlejawa — APPROVED** [review]
LGTM
Thanks!

**Comment by bergwolf** [comment]
Somehow CI is not using the CNI plugins installed by script?

> 11:32:00 Install CNI plugins
11:32:00 Retrieve CNI plugins repository
11:32:00 /tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-22.04-PR/go/src /tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-22.04-PR/go/src/github.com/kata-containers/tests
11:32:00 Cloning into 'plugins'...
11:32:00 Note: switching to 'v1.1.1'.


> 11:35:59 time="2022-10-28T03:35:58.935144344Z" level=error msg="Failed to destroy network for sandbox \"d476039156f9a6fb7e879e9f91ffc88836fa1371b76ea5672afde99d0f56c378\"" error="plugin type=\"portmap\" failed (delete): incompatible CNI versions; config is \"1.0.0\", plugin supports [\"0.1.0\" \"0.2.0\" \"0.3.0\" \"0.3.1\" \"0.4.0\"]"

**Comment by liubin** [comment]
This PR depends on https://github.com/kata-containers/kata-containers/pull/5511, but at line 165, it will install the golang version from the main branch, at this point, it's 1.17.3 for now.

https://github.com/kata-containers/tests/blob/2c02eda42cc0b509dba1d99e19500a161ff34039/.ci/jenkins_job_build.sh#L161-L176

Golang installed after line 175 will get the desired version 1.19.

This leads [s390x fail](http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-s390x-unit-PR/241/consoleFull) due to `"ERROR: golang minor version too old: got 1.17.3, need atleast 1.19"`.

**Comment by UnmeshDeodhar** [comment]
Without the go version update we are not able to run coco operator CI and coco general CI on the same node. Please let us know if we can do anything to help push this along. @fidencio @bergwolf @liubin

**Review by liubin — APPROVED** [review]

**Review by fidencio — APPROVED** [review]
lgtm, thanks @bergwolf!
